woocommerce/woocommerce-ios
# Configurations for Product Selector for customization in New Order and Edit Coupon flows

Part of #6489 


### Description

As part of the Coupons project, we want to show a list of products to select/exclude from coupons. This PR adds basic configurations for the `ProductSelector` view to make it reusable and customizable between Order Creation and Coupon Edit/Creation flows.

Changes:
- Add a new struct named `Configuration` to hold customized details for the product selector view.
- Move the configs for the Order Creation flow to a private extension in New Order.
- Add new configs for Select Products and Exclude Products in the coupon flows. These configs are used in the AddEditCoupon view and CouponRestrictions view.

### Testing instructions

Testing New Order:
- Navigate to the Orders tab, select "+" button and proceed to Create new order.
- Select "+ Add Product", and notice that the product selector view is displayed. The view should looked and worked properly as before.

Testing Edit Coupon:
- Make sure that your test store has coupons enabled and has set up at least one coupon.
- Navigate to the Menu tab, select Coupons.
- Select a coupon with a supported discount type > select the ellipsis menu > proceed to Edit Coupon.
- On the edit screen, select Edit Products. Notice that a new screen titled "Select Products" shows up. The selection of items on this screen will be updated in a next PR.
- Navigate back to the Edit Coupon screen. Scroll down to the bottom and select Usage Restrictions > Exclude Products. Notice that a new screen titled "Exclude Products" shows up. The selection of items on this screen will be updated in a next PR.

### Screenshots

Order creation:
https://user-images.githubusercontent.com/5533851/164146384-4d54ad05-1e6e-41ed-9165-200ec8b06b2c.mp4

Edit Coupon:
https://user-images.githubusercontent.com/5533851/164146461-e0c8efd3-2869-4356-8e0f-dca193f435d4.mp4


---
